### PR TITLE
Refactor of ether send

### LINKED ISSUE #516: False positive on ether_send analysis
## Description

Withdrawal of Ether is wrongly reported in the following contract:

```
contract Crowdfunding {

  mapping(address => uint) public balances;
  address public owner;
  uint256 INVEST_MIN = 1 ether;
  uint256 INVEST_MAX = 10 ether;

  modifier onlyOwner() {
    require(msg.sender == owner);
    _;
  }

  function Crowdfunding() {
    owner = msg.sender;
  }

  function withdrawfunds() onlyOwner {
    msg.sender.transfer(this.balance);
  }

  function invest() public payable {
    require(msg.value > INVEST_MIN && msg.value < INVEST_MAX);

    balances[msg.sender] += msg.value;
  }

  function getBalance() public constant returns (uint) {
    return balances[msg.sender];
  }

}
```
## How to Reproduce

```
$MYTH -x constructor_correctly_named.sol 
==== Ether send ====
Type: Warning
Contract: Crowdfunding
Function name: withdrawfunds()
PC address: 713
In the function `withdrawfunds()` a non-zero amount of Ether is sent to msg.sender.
It seems that this function can be called without restrictions.
--------------------
In file: constructor_correctly_named.sol:19

msg.sender.transfer(this.balance)

--------------------
```

## Expected behavior

"Ether send" should not be reported here because the `onlyOwner` checks the caller against the owner, and owner is correctly initialized during construction.

## Environment

Please modify for your setup

- Mythril version: Master
- Solidity compiler and version:  0.4.24+commit.e67f0147.Darwin.appleclang
- Python version: Python 3.6.3
- OS and Version: Mac OS High Sierra

## Additional Environment or Context

Opposite of #515 :)

### PR DESCRIPTION
**Pull request by JoranHonig**
Refactors the ethersend to work similar to the new suicide detection module.

fixes #516

### CODE REVIEW

File: tests/testdata/outputs_expected/multi_contracts.sol.o.text [hunk 1]
```diff
@@ -4,7 +4,6 @@ Type: Warning
 Contract: Unknown
 Function name: _function_0x8a4068dd
 PC address: 142
-In the function `_function_0x8a4068dd` a non-zero amount of Ether is sent to msg.sender.
-It seems that this function can be called without restrictions.
+It seems that an attacker is able to execute an call instruction, this can mean that the attacker is able is able to extract funds out of the contract.
```
Comment (norhh) [inline tests/testdata/outputs_expected/multi_contracts.sol.o.text]: can you remove the multiple "is able" from "is able is able"

File: tests/testdata/outputs_expected/multi_contracts.sol.o.markdown [hunk 2]
```diff
@@ -9,5 +9,4 @@
 
 ### Description
 
-In the function `_function_0x8a4068dd` a non-zero amount of Ether is sent to msg.sender.
-It seems that this function can be called without restrictions.
+It seems that an attacker is able to execute an call instruction, this can mean that the attacker is able is able to extract funds out of the contract.
```
Comment (norhh) [inline tests/testdata/outputs_expected/multi_contracts.sol.o.markdown]: same here, multiple "is able"

File: tests/testdata/outputs_expected/multi_contracts.sol.o.text (hunk 1 above)
Comment (norhh) [reply tests/testdata/outputs_expected/multi_contracts.sol.o.text]: not exactly here but at the source :sweat_smile:

**Comment by b-mueller** [comment]
Seems like one of the ether_send samples is not detected:

```
$ $MYTH -x tests/testdata/input_contracts/ether_send.sol 
The analysis was completed successfully. No issues were detected.

```

**Comment by b-mueller** [comment]
.... hopefully I didn't mess up the code when resolving conflicts :/

**Pull request closed without merging**